incubator/riemann/templates/deployment.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}"
+        imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+        {{- if .Values.service.ports.tcp.internal }}
```

Comment:
Probes should not depend on the service. In fact, your service probably won't even work because the container does not expose any ports.

Reply:
Thanks! I got cute about using the values here, you're right. Those should be container ports, not service ports.

Also, the ports are already exposed via the Dockerfile in the image, which is why this works. Is it expected to have containerPort specified in any case? Easy to add.

Reply:
I think it'd be better if the container exposed ports. You could use named ports and reference them.

Here's an example:
https://github.com/kubernetes/charts/blob/master/stable/mailhog/templates/deployment.yaml#L39-L52
https://github.com/kubernetes/charts/blob/master/stable/mailhog/templates/service.yaml#L28-L42

Reply:
I did add the explicit containerPorts.

However, I'm not sure I understand the "named ports" comment. Are you referring to IANA Well-Known ports? Or is there some means of naming and referencing ports in Kubernetes? I'm finding some documentation that seems to hint at that being true. If I name a container port I can reference it by name in the service targetPort?